src/communicate.rs

```diff
@@ -212,20 +223,30 @@ pub fn handle_wire_msg(peer_addr: SocketAddr, wire_msg: WireMsg) {
                             pending_reads.push(wire_msg);
                         }
                         ToPeer::NotNeeded => dispatch_wire_msg(
-                            // Message from a client
-                            peer_addr,
-                            q_conn,
-                            c.our_ext_addr_tx.take(),
-                            &c.event_tx,
-                            wire_msg,
-                        ),
-                        ToPeer::Established { ref q_conn, .. } => dispatch_wire_msg(
-                            peer_addr,
+                            Peer::Client { peer_addr },
                             q_conn,
                             c.our_ext_addr_tx.take(),
                             &c.event_tx,
                             wire_msg,
+                            &mut c.bootstrap_cache,
```

Comment:
This is wrong - if `ToPeer` is not-needed then that peer is basically a client. Why would we want to add that to the bootstrap cache ?

Reply:
It won't be added to the cache - the code is just there to confirm with the `dispatch_wire_msg()` signature.
```
                        ToPeer::NotNeeded => dispatch_wire_msg(
                            Peer::Client { peer_addr },
                            q_conn,
                            c.our_ext_addr_tx.take(),
                            &c.event_tx,
                            wire_msg,
                            &mut c.bootstrap_cache,
                        ),
```
See we are telling `dispatch_wire_msg` that this is the `Client` peer and it will act accordingly.